### PR TITLE
IMPRV/DataSourceOrigin:FindRowByIndex

### PR DESCRIPTION
**Pull request by Dwaynekj**
Dependent on merging https://github.com/openfin/fin-hypergrid/pull/367

### CODE REVIEW

File: src/dataSources/DataSourceOrigin.js [hunk 1]
```diff
@@ -170,6 +170,47 @@ var DataSourceOrigin = DataSourceBase.extend('DataSourceOrigin',  {
...
+        var result;
+
+        if (arguments.length < 1) {
+            throw 'Expected at least 1 argument but found ' + arguments.length + '.';
+        }
+
+        if (typeof index !== 'number') {
+            throw 'Expected at index to be a number but got ' + index + '.';
+        }
+        result = this.data.find(function(row, idx) {
```
Comment (joneit) [inline src/dataSources/DataSourceOrigin.js]: No! Use `getDataIndex` here. Explanation: We need to call getDataIndex to transpose all the indexes. In other words, if the filtered & sorted dataset has 10 rows left out of 1000 and we want to delete row the 7th row, calling `getDataIndex(7-1)` will return the underlying index of that row in `source.data`. Get it?

File: src/dataSources/DataSourceOrigin.js [hunk 2]
```diff
@@ -170,6 +170,47 @@ var DataSourceOrigin = DataSourceBase.extend('DataSourceOrigin',  {
...
+            if (!row) { return; }
+            return index === idx;
+        });
+
+        if (result) {
+            if (replacement === null) {
+                this.data.splice(index, 1);
+            } else if (typeof replacement === 'object') {
+                this.data[index] = replacement;
+            } else if (replacement !== undefined) {
```
Comment (joneit) [inline src/dataSources/DataSourceOrigin.js]: Replace this line with:
```javascript
            } else if (replacement === undefined && arguments.length >= 2) {
                delete this.data[index];
            } else {
```
Explanation: My comment was wrong. `undefined` should delete the row element. This leaves `undefined` in the element which renders as a blank row.

File: src/dataSources/DataSourceOrigin.js (hunk 1 above)
Comment (Dwaynekj) [inline src/dataSources/DataSourceOrigin.js]: I think we should encourage the application developer to implement what your describing. However at DataSourceOrigin base we should allow the ability to delete the underlying data. Thoughts?

File: src/dataSources/DataSourceOrigin.js (hunk 2 above)
Comment (Dwaynekj) [inline src/dataSources/DataSourceOrigin.js]: Done

File: src/dataSources/DataSourceOrigin.js (hunk 1 above)
Comment (joneit) [inline src/dataSources/DataSourceOrigin.js]: as discussed we'll leave calling `getDataIndex` to the caller to pre-translate his index if that's what he wants to do... **but hey DJ you don't need `.find(...)` here! just do `this.data[index]`, right?**

Comment (joneit) [inline src/dataSources/DataSourceOrigin.js]: wow I have a better idea. if we put `getDataIndex` in the function, then people can still call it both ways:
```
dataModel.source.findRowByIndex(n, null) // deletes actual row n
dataModel.dataSource.findRowByIndex(n, null) // deletes transformed (displayed) row n
```

Comment (Dwaynekj) [inline src/dataSources/DataSourceOrigin.js]: oh duh! Good Catch

**Pull request merged**